Engine/Module/ModuleScriptManager.cpp

```diff
@@ -0,0 +1,339 @@
+		if (script_func != nullptr)
+		{
+			Script* script = script_func();
+			script->AddReferences(owner, App);
+			return script;
+		}
+		return nullptr;
+	}
+	return nullptr;
+}
+
+ComponentScript* ModuleScriptManager::CreateComponentScript()
+{
+	ComponentScript* new_script = new ComponentScript();
+	scripts.push_back(new_script);
+	return new_script;
+}
+
+void ModuleScriptManager::RemoveComponentScript(ComponentScript * script_to_remove)
+{
+	auto it = std::find(scripts.begin(), scripts.end(), script_to_remove);
+	if (it != scripts.end())
+	{
+		delete *it;
+		scripts.erase(it);
+	}
+}
+
+void ModuleScriptManager::LoadScriptList() 
+{
+	if (scripts_list.size() > 0) 
+	{
+		scripts_list.clear();
+	}
+	
+	size_t readed_bytes;
+	char* scripts_file_data = App->filesystem->Load(SCRIPT_LIST_PATH, readed_bytes);
+	if (scripts_file_data != nullptr)
+	{
+		std::string serialized_scripts_string = scripts_file_data;
+		free(scripts_file_data);
+
+		Config scripts_config(serialized_scripts_string);
+
+		std::vector<Config> scripts_list_configs;
+		scripts_config.GetChildrenConfig("Scripts", scripts_list_configs);
+		for (unsigned int i = 0; i < scripts_list_configs.size(); ++i)
+		{
+			scripts_list.push_back(scripts_list_configs[i].config_document.GetString());
+		}
+	}
+
+}
+
+void ModuleScriptManager::RunScripts()
+{
+	for (auto &component_script : scripts)
+	{
+		component_script->Update();
+	}
+}
+
+void ModuleScriptManager::RemoveScriptPointers()
+{
+	for (auto &component_script : scripts)
+	{
+		component_script->script = nullptr;
+	}
+}
+
+void ModuleScriptManager::initDLL()
+{
+	patchDLL(SCRIPTS_DLL_PATH, working_directory.c_str());
+	gameplay_dll = LoadLibrary(SCRIPT_DLL_FILE);
+}
+
+void ModuleScriptManager::ReloadDLL() 
+{
+	if (gameplay_dll != nullptr) 
+	{
+		if (!FreeLibrary(gameplay_dll)) 
+		{
+			return;
+		}
+		else 
+		{
+			RemoveScriptPointers();
+			remove(SCRIPT_DLL_FILE);
+		}
+	}
+	initDLL();
+	InitResourceScript();
+
+}
+
+void ModuleScriptManager::GetCurrentPath() 
+{
+	TCHAR NPath[MAX_PATH];
+	GetCurrentDirectory(MAX_PATH, NPath);
+	working_directory = NPath;
+	working_directory += "/GamePlaySyste_.dll";
+}
+
+size_t ModuleScriptManager::CStrlastIndexOfChar(const char* str, char find_char)
+{
+	intptr_t i = strlen(str) - 1;
+	while (i >= 0)
+	{
+		if (str[i] == find_char) 
+		{
+			return i;
+		}	
+		--i;
+	}
+	return (size_t)-1;
+}
+
+bool ModuleScriptManager::patchFileName(char* filename)
+{
+	size_t	dot_idx = CStrlastIndexOfChar(filename, '.');
+	if (dot_idx != (size_t)-1)
+	{
+		filename[dot_idx - 1] = '_';
+		return true;
+	}
+	else 
+	{
+		return false;
+	}
+		
+}
+
+bool ModuleScriptManager::CopyPDB(const char* source_file, const char* destination_file, bool overwrite_existing)
+{
+	return CopyFile(source_file, destination_file, !overwrite_existing);
+}
+
+bool ModuleScriptManager::patchDLL(const char* dll_path, const char patched_dll_path[MAX_PATH])
```

Comment:
If it is an array if chars maybe should be passed by reference?
I am not sure though

Reply:
I modified the second argument, now is a pointer.